app/uk/gov/hmrc/apisubscriptionfields/service/SubscriptionFieldsService.scala

```diff
@@ -36,22 +36,22 @@ class SubscriptionFieldsService @Inject()(repository: SubscriptionFieldsReposito
                                           uuidCreator: UUIDCreator) {
 
   def upsert(clientId: ClientId, apiContext: ApiContext, apiVersion: ApiVersion, subscriptionFields: Fields): Future[(SubscriptionFieldsResponse, Boolean)] = {
+
     def update(existingFieldsId: UUID): Future[SubscriptionFieldsResponse] =
       save(SubscriptionFields(clientId.value, apiContext.value, apiVersion.value, existingFieldsId, subscriptionFields))
 
     def create(): Future[SubscriptionFieldsResponse] =
       save(SubscriptionFields(clientId.value, apiContext.value, apiVersion.value, uuidCreator.uuid(), subscriptionFields))
 
     Logger.debug(s"[upsert subscription fields] clientId: $clientId, apiContext: $apiVersion, apiVersion: $apiVersion")
-
-    repository.fetch(clientId, apiContext, apiVersion) flatMap {
-      o =>
-        o.fold(
-          create() map { x => (x, true) }
-        )(
-          existing => update(existing.fieldsId) map { x => (x, false) }
-        )
```

Comment:
I would recommend pushing this sort of logic into the Repository, and extracted if possible to the `MongoCrudHelper`. Also this logic may not be necessary as we get information about whether operation resulted in insert or update from the  `WriteResult`